packages/flutter_tools/lib/src/run_hot.dart

```diff
@@ -1064,16 +1069,33 @@ class ProjectFileInvalidator {
       fs.file(packagesPath).uri,
     ];
     final List<Uri> invalidatedFiles = <Uri>[];
-    for (final Uri uri in urisToScan) {
-      final DateTime updatedAt = fs.statSync(
-          uri.toFilePath(windows: platform.isWindows)).modified;
-      if (updatedAt != null && updatedAt.isAfter(lastCompiled)) {
-        invalidatedFiles.add(uri);
+
+    if (asyncScanning) {
+      final List<Future<void>> waitList = <Future<void>>[];
+      for (final Uri uri in urisToScan) {
+        waitList.add(fs
+            .stat(uri.toFilePath(windows: platform.isWindows))
```

Comment:
@zanderso does the async stat call have the same problem with limited file descriptors that reading/writing does?

Reply:
No, it doesn't have exactly the same problem, however there are two considerations:
1. On Windows, a handle will be briefly opened if the entity is a reparse point. This will only be a problem if this code is going to be encountering a directory containing 10,000+ reparse points.
2. The VM will spawn up to 32 threads per isolate to service async io requests. If this code will be going above that, then it probably makes sense to use a pool (package:pool) even if file descriptor/handle exhaustion is not an issue.